umbrella.d.ts

```diff
@@ -1,207 +1,1484 @@
+     * issue.](https://github.com/franciscop/umbrella/issues/114)
+     *
+     * **Note:** multiple-select are not supported in Internet Explorer, [similarly to
+     * jQuery.](https://github.com/jquery-archive/jquery-mobile/issues/3947)
+     *
+     * @example
+     *   // For this form:
+     *   <form action='/contact' method='POST'>
+     *     Email:
+     *     <input type='email' name='email' value='test@example.com' />
+     *     Message:
+     *     <textarea type='email' name='message'>
+     *       Hello world
+     *     </textarea>
+     *     <button>Send</button>
+     *   </form>;
+     *
+     *   // When the user clicks on the "Send" button, the following handler can be used to send the data through Ajax:
+     *   // .handle() == .on() + preventDefault()
+     *   u('form.contact').handle('submit', async event => {
+     *     // Body: email=test@example.com&message=Hello+world
+     *     const body = u(event.target).serialize();
+     *     const data = await fetch('/contact', {
+     *       method: 'POST',
+     *       body,
+     *     }).then(res => res.json());
+     *     console.log('Response data:', data);
+     *   });
+     *
+     *   // If you were using the native FormData:
+     *   // .handle() == .on() + preventDefault()
+     *   u('form.contact').handle('submit', async event => {
+     *     const body = new FormData(event.target);
+     *     const data = await fetch('/contact', {
+     *       method: 'POST',
+     *       body,
+     *     }).then(res => res.json());
+     *     console.log('Response data:', data);
+     *   });
+     *
+     * @returns {string} - Returns the instance of Umbrella for method chaining.
+     */
     serialize(): string;
 
-    siblings: typeof u.children;
+    /**
+     * Get the siblings of all the nodes with an optional filter.
+     *
+     * **The signature is the same as the {@linkcode children} method.**
+     *
+     * [Documentation](https://umbrellajs.com/documentation#siblings)
+     *
+     * @example
+     *   // Get the all the siblings of the hovered <li>
+     *   u('li:hover').siblings('li:first-child');
+     *
+     *   // Get all the siblings
+     *   u('li').siblings();
+     *
+     * @type {this['children']}
+     */
+    siblings: Umbrella<NODE>['children'];
 
+    /**
+     * Get the [bounding client rect](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect) of
+     * the first matched element. This has height, width, top, left, right and bottom properties
+     *
+     * [Documentation](https://umbrellajs.com/documentation#size)
+     *
+     * @example
+     *   // Get the size and position of the first matched <div> element
+     *   const size = u('div').size();
+     *   console.log(`Width: ${size.width}, Height: ${size.height}`);
+     *
+     * @returns {DOMRect} - A DOMRect object containing the size and position of the first element.
+     */
     size(): DOMRect;
```

Comment:
[Judging by code](https://github.com/franciscop/umbrella/blob/7988c20815ae966d3603cbc684a1580107f711e5/src/plugins/size/size.js#L3), this method can throw an exception, when the Umbrella instance doesn't have any nodes